lib/rules/typescript-uses-vars.js

```diff
@@ -0,0 +1,154 @@
+/**
```

Comment:
I based that on the `jsx-uses-vars` "rule" in the react plugin.

I am putting rule in quotes because it is not really a rule at all. All the code is doing is marking things as used; it doesn't define a set of behaviour that the source code must implement, like the `no-unused-vars` core rule does.

I am totally happy to go with your final decision on the name, but that was the reasoning :)